src/main/scala/sparkz/core/network/SendingStrategy.scala

```diff
@@ -31,6 +33,18 @@ case object BroadcastTransaction extends SendingStrategy {
   }
 }
 
+case class BroadcastBlock(settings: NetworkSettings) extends SendingStrategy {
+  val maxBlockBroadcastPeers: Int = settings.maxForgerConnections +
+    (settings.maxOutgoingConnections + settings.maxIncomingConnections) / 2
+
+  override def choose(peers: Seq[ConnectedPeer]): Seq[ConnectedPeer] = {
+    val forgerPeers = peers.filter(_.peerInfo.exists(_.peerSpec.forgerPeer))
```

Comment:
Maybe here we can use the `partition` method to get the two sequences at once, something like: `val (forgerPeers, remainingPeers) = peers.partition(_.peerInfo.exists(_.peerSpec.forgerPeer))`